source/misc/bash_startup.in

```diff
@@ -1,257 +1,95 @@
-#!/bin/bash
-# This is based on "preexec.bash" but is customized for iTerm2.
+# This is inspired by "preexec.bash" but is customized for iTerm2. Sourced, so
+# no shebang. This module requires 2 bash features you must not otherwise be
+# using: the DEBUG trap and the PROMPT_COMMAND variable.
 
-# Note: this module requires 2 bash features which you must not otherwise be
-# using: the "DEBUG" trap, and the "PROMPT_COMMAND" variable.  iterm2_preexec_install
-# will override these and if you override one or the other this _will_ break.
+osc='\033]'; cso='\007' # Beginning, ending escape sequences.
 
-# This is known to support bash3, as well as *mostly* support bash2.05b.  It
-# has been tested with the default shells on MacOS X 10.4 "Tiger", Ubuntu 5.10
-# "Breezy Badger", Ubuntu 6.06 "Dapper Drake", and Ubuntu 6.10 "Edgy Eft".
+# tmux / screen are not supported; I think this is the theoretical tmux hack?
+[ $TERM = screen ] && osc='\033Ptmux;\033\033]' && cso='\007\033\\'
 
+orig_ps1=$PS1; prev_ps1=$PS1 # orig is the user-set PS1; prev is our last PS1.
+preexec_interactive_mode= # If set: "just executed prompt; waiting for input."
 
-# Copy screen-run variables from the remote host, if they're available.
+[ -z "$iterm2_hostname" ] && iterm2_hostname=$(uname -n) # hostname -f == slow
 
-# Saved copy of your PS1. This is used to detect if the user changes PS1
-# directly. prev_ps1 will hold the last value that this script set PS1 to
-# (including various custom escape sequences). orig_ps1 always holds the last
-# user-set value of PS1.
-orig_ps1="$PS1"
-prev_ps1="$PS1"
+# Usage: iterm2_set_user_var key value
+iterm2_set_user_var(){
+  var=$1; shift; val="$*"
+  printf $osc'1337;SetUserVar=%s=%s'$cso "$var" "$val"
+  unset var; unset val
+}
 
-# This variable describes whether we are currently in "interactive mode";
-# i.e. whether this shell has just executed a prompt and is waiting for user
-# input.  It documents whether the current command invoked by the trace hook is
-# run interactively by the user; it's set immediately after the prompt hook,
-# and unset as soon as the trace hook is run.
-preexec_interactive_mode=""
+# Users can write their own version of this method. It should call
+# iterm2_set_user_var but not produce any other output. # [ Huh? ]
```

Comment:
I see. To me `iterm2_print_user_vars` sounded like the opposite of `iterm2_get_user_vars`, i.e., how you'd access these variables. I'm having trouble with `\(var)`... but I'll play with it some more, I'm probably not doing it right.